sentry/src/main/java/io/sentry/protocol/SentryTransaction.java

```diff
@@ -90,15 +91,16 @@ public SentryTransaction(final @NotNull SentryTracer sentryTracer) {
             tracerContext.getDescription(),
             tracerContext.getSamplingDecision(),
             tracerContext.getStatus(),
-            tracerContext.getOrigin()));
+            tracerContext.getOrigin());
+
+    contexts.setTrace(tracerContextToSend);
     for (final Map.Entry<String, String> tag : tracerContext.getTags().entrySet()) {
       this.setTag(tag.getKey(), tag.getValue());
     }
 
-    final Map<String, Object> data = sentryTracer.getData();
     if (data != null) {
       for (final Map.Entry<String, Object> tag : data.entrySet()) {
-        this.setExtra(tag.getKey(), tag.getValue());
+        tracerContextToSend.setData(tag.getKey(), tag.getValue());
```

Comment:
We're relying on `contexts.setTrace` keeping the reference which we can modify after setting it. I'd prefer if we add `data` entries before calling `setTrace`.